pkg/capabilities/capabilities.go

```diff
@@ -1,88 +1,295 @@
 package capabilities
 
 import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/aquasecurity/tracee/pkg/logger"
 	"kernel.org/pub/linux/libs/security/libcap/cap"
 )
 
-// CheckRequired check if given capabilities include all required capabilities in the effective set.
-// If the error is of the MissingCapabilitiesError type, it means some capabilities are missing.
-// Other errors might rise because of capabilities extraction failure, or internal errors.
-func CheckRequired(reqCaps []cap.Value) error {
-	selfCaps, err := getCurrentProcessCapabilities()
+const pkgName = "capabilities"
+
+//
+// "Effective" might be at protection rings 0,1,2,3
+// "Permitted" is always at ring0 (so effective can migrate rings)
+// "Bound" will bet set to unprivileged so exec() can't inherit capabilities.
+//
+
+type ringType int
+
+const (
+	Privileged   ringType = iota // ring0 (all capabilities enabled, startup/shutdown)
+	Required                     // ring1 (needed capabilities only: config time)
+	Requested                    // ring2 (temporary specific capabilities)
+	Unprivileged                 // ring3 (no capabilities: runtime)
+)
+
+type Capabilities struct {
+	have   *cap.Set
+	all    map[cap.Value]map[ringType]bool
+	bypass bool
+}
+
+func NewCapabilities(bypass bool) *Capabilities {
+	c := &Capabilities{}
+	c.initialize(bypass)
+	return c
+}
+
+func (c *Capabilities) initialize(bypass bool) {
```

Comment:
If this can fail (internally it panics if yes so it can), I think it would be better to keep to our conventions and return an error here. The constructor `NewCapabilities` would likewise have an error which we would return from tracee.

Reply:
I'm ok with that change.

Reply:
I don't think not being able to handle capabilities is a recoverable error, so I haven't returned any error and panic'ed. Closing this comment and keeping the other opened if you have more comments on this.